ci/scripts/rust_clippy.sh

```diff
@@ -19,3 +19,7 @@
 
 set -ex
 cargo clippy --all-targets --workspace --features avro,pyarrow,integration-tests,extended_tests -- -D warnings
+
+# Update packages incrementally for stricter Clippy checks
+# TODO: add tracking issue for the remaining workspace packages like `datafusion-catalog`
```

Comment:
I think this would be better to add at the module level (rather than in the CI script) so that it would also be flagged locally when people ran clippy.

Similar to this:

https://github.com/apache/datafusion/blob/e4f2b4962076c3497b2689e8773bac59dc0b21a5/datafusion/common/src/lib.rs#L25